sklearn/tree/_splitter.pyx

```diff
@@ -923,16 +918,13 @@ cdef class BaseSparseSplitter(Splitter):
             return zero_pos
 
         while p < partition_end:
-            value = Xf[p]
```

Comment:
Some of these were here because of some odd python interactions Cython creates. Could you please either check the generated C code, or do benchmarks to see how it behaves?

Reply:
the difference in c is here 
```
--- __pyx_t_7sklearn_4tree_5_tree_SIZE_t __pyx_v_p;
--- __pyx_v_value = (__pyx_v_Xf[__pyx_v_p]);
---  __pyx_t_3 = ((__pyx_v_value <= __pyx_v_threshold) != 0);
+++  __pyx_t_3 = (((__pyx_v_Xf[__pyx_v_p]) <= __pyx_v_threshold) != 0);

 if (__pyx_t_3) {
```
```